p2p/peerResend.go

```diff
@@ -0,0 +1,99 @@
+package p2p
+
+import (
+	"crypto/sha1"
+	"sync"
+	"time"
+)
+
+type PeerResend struct {
+	mtx      sync.RWMutex
+	buckets  []*PRBucket
+	pc       int
+	interval time.Duration
+	stopper  chan interface{}
+}
+
+type PRBucket struct {
+	mtx  sync.RWMutex
+	data map[[sha1.Size]byte]bool
+}
+
+func (b *PRBucket) Has(hash [sha1.Size]byte) bool {
+	b.mtx.RLock()
+	defer b.mtx.RUnlock()
+	return b.data[hash]
+}
+
+func (b *PRBucket) Add(hash [sha1.Size]byte) {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+	b.data[hash] = true
+}
+
+func NewPeerResend(buckets int, interval time.Duration) *PeerResend {
+	pr := new(PeerResend)
+	pr.interval = interval
```

Comment:
a detail, but it looks like you don't really need to store the value of interval, you could just pass it as an argument to Cleanup and forget about it. Passing it explicitly as an argument of Cleanup has the added advantage of readability as it is clear it's the cleanup interval and not something else

Reply:
That's true. I'm also unexporting Cleanup as that shouldn't be exported.